src/sortStudents.ts

```diff
@@ -1,16 +1,109 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(students: Student[],
+  sortBy: SortType, order: SortOrder): Student[] {
+  function averGrade(arr: number[]): number {
+    const grade = arr.reduce((sum: number, a: number):
+    number => sum + a, 0) / arr.length;
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+    return grade;
+  }
+
+  const newStudents = [...students];
+
+  switch (sortBy) {
+    case SortType.Name: {
+      if (order === 'asc') {
+        newStudents.sort((a: Student, b: Student):
+        number => a.name.localeCompare(b.name));
+      } else {
+        newStudents.sort((a: Student, b: Student):
+        number => b.name.localeCompare(a.name));
+      }
+
+      break;
+    }
+
+    case SortType.Surname: {
+      if (order === 'asc') {
+        newStudents.sort((a: Student, b: Student):
+        number => a.surname.localeCompare(b.surname));
+      } else {
+        newStudents.sort((a: Student, b: Student):
+        number => b.surname.localeCompare(a.surname));
+      }
+
+      break;
+    }
```

Comment:
одинаковый тип по которому ищешь, объедини
```suggestion
    case SortType.Name: 
    case SortType.Surname: 
      if (order === 'asc') {
        newStudents.sort((a: Student, b: Student):
        number => a.surname.localeCompare(b.surname));
      } else {
        newStudents.sort((a: Student, b: Student):
        number => b.surname.localeCompare(a.surname));
      }

      break;
```